drivers/adcxx1c/adcxx1c.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2017 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_adcxx1x
+ *
+ * @file
+ * @brief       ADCXX1C ADC device driver
+ *
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ */
+
+#include "adcxx1c.h"
+#include "adcxx1c_params.h"
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#define I2C_SPEED I2C_SPEED_FAST
+
+#define I2C (dev->params.i2c)
+#define ADDR (dev->params.addr)
+
+#define ADCXX1C_CONV_RES_ADDR      (0)
+#define ADCXX1C_ALERT_STATUS_ADDR  (1)
+#define ADCXX1C_CONF_ADDR          (2)
+#define ADCXX1C_LOW_LIMIT_ADDR     (3)
+#define ADCXX1C_HIGH_LIMIT_ADDR    (4)
+#define ADCXX1C_HYSTERESIS_ADDR    (5)
+#define ADCXX1C_LOWEST_CONV_ADDR   (6)
+#define ADCXX1C_HIGHEST_CONV_ADDR  (7)
+
+#define ADCXX1C_CONF_ALERT_PIN_EN   (1 << 2)
+#define ADCXX1C_CONF_ALERT_FLAG_EN  (1 << 3)
+
+int adcxx1c_init(adcxx1c_t *dev, const adcxx1c_params_t *params)
+{
+    assert(dev && params);
+
+    dev->params = *params;
```

Comment:
sure this works? De-referencing a pointer and then accessing it's members as `dev->params.XX`?